pkg/auth/user/UserService.go

```diff
@@ -944,15 +949,103 @@ func (impl *UserServiceImpl) GetAll() ([]bean.UserInfo, error) {
+	// get req from arguments
+	request := impl.getRequestWithFiltersArgsOrDefault(sortOrder, sortBy, offset, totalSize, showAll, searchKey)
+	if request.ShowAll {
+		return impl.getAllDetailedUsers()
+	}
+	// Setting size as zero to calculate the total number of results based on request
+	size := request.Size
+	request.Size = 0
+
+	// Build query from query builder
+	query := impl.userListingRepositoryQueryBuilder.GetQueryForUserListingWithFilters(request)
+	totalCount, err := impl.userRepository.GetCountExecutingQuery(query)
 	if err != nil {
 		impl.logger.Errorw("error while fetching user from db", "error", err)
 		return nil, err
 	}
+	request.Size = size
+
+	query = impl.userListingRepositoryQueryBuilder.GetQueryForUserListingWithFilters(request)
+	models, err := impl.userRepository.GetAllExecutingQuery(query)
+	if err != nil {
+		impl.logger.Errorw("error while fetching user from db", "error", err)
+		return nil, err
+	}
+
+	response, err := impl.getUserResponse(models)
+	if err != nil {
+		impl.logger.Errorw("error in getUserResponseWithLoginAudit", "err", err)
+		return nil, err
+	}
+
+	listingResponse := &bean.UserListingResponse{
+		Users:      response,
+		TotalCount: totalCount,
+	}
+	return listingResponse, nil
+
+}
+
+func (impl UserServiceImpl) getRequestWithFiltersArgsOrDefault(sortOrder string, sortBy string, offset int, totalSize int, showAll bool, searchKey string) *bean.FetchListingRequest {
+	sortByRes, size := impl.userCommonService.GetDefaultValuesIfNotPresent(sortBy, totalSize, false)
+	request := &bean.FetchListingRequest{
+		SortOrder: bean2.SortOrder(sortOrder),
+		SortBy:    sortByRes,
+		Offset:    offset,
+		Size:      size,
+		ShowAll:   showAll,
+		SearchKey: searchKey,
+	}
+	return request
+}
+
+func (impl UserServiceImpl) getAllDetailedUsers() (*bean.UserListingResponse, error) {
+	response, err := impl.GetAllDetailedUsers()
+	if err != nil {
+		impl.logger.Errorw("error in getAllDetailedUsers", "err", err)
+		return nil, err
+	}
+
+	listingResponse := &bean.UserListingResponse{
+		Users:      response,
+		TotalCount: len(response),
+	}
+	return listingResponse, err
+}
+
+func (impl UserServiceImpl) getUserResponse(model []repository.UserModel) ([]bean.UserInfo, error) {
+	var response []bean.UserInfo
+	for _, m := range model {
+		lastLoginTime := impl.getLastLoginTime(m)
+		response = append(response, bean.UserInfo{
+			Id:            m.Id,
+			EmailId:       m.EmailId,
+			RoleFilters:   make([]bean.RoleFilter, 0),
+			Groups:        make([]string, 0),
+			LastLoginTime: lastLoginTime,
+		})
+	}
+	if len(response) == 0 {
+		response = make([]bean.UserInfo, 0)
+	}
+	return response, nil
+}
+
+func (impl *UserServiceImpl) GetAllDetailedUsers() ([]bean.UserInfo, error) {
+	query := impl.userListingRepositoryQueryBuilder.GetQueryForAllUserWithAudit()
```

Comment:
this query can be modified in gorm.

Reply:
can't be done as not collecting user audit in user model, which have to be ignored for previous methods, as it will be considered a column acc to gorm.